jobs/JGDAS_FIT2OBS

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/env bash
+
+source "${HOMEgfs}/ush/preamble.sh"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "fit2obs" -c "base fit2obs"
+
+
+##############################################
+# Set variables used in the script
+##############################################
+
+export CDATE=${CDATE:-${PDY}${cyc}}
+export CDUMP=${CDUMP:-${RUN:-"gdas"}}
+
+export xdate=$(${NDATE} -${VBACKUP_FITS} ${CDATE})
+export CDATE=${xdate} # Reset CDATE to xdate for lookback
+export vday=$(echo ${xdate} | cut -c1-8)
+export vcyc=$(echo ${xdate} | cut -c9-10)
+
+export COM_INA=${ROTDIR}/gdas.${vday}/${vcyc}/atmos
+export COM_INF='$ROTDIR/vrfyarch/gfs.$fdy/$fzz'
+export COM_PRP=${ROTDIR}/gdas.${vday}/${vcyc}/obs
+
+export PRPI=${COM_PRP}/${CDUMP}.t${vcyc}z.prepbufr
+export sig1=${COM_INA}/${CDUMP}.t${vcyc}z.atmanl.nc
+export sfc1=${COM_INA}/${CDUMP}.t${vcyc}z.atmanl.nc
+export CNVS=${COM_INA}/${CDUMP}.t${vcyc}z.cnvstat
+
+export OUTPUT_FILETYPE=${OUTPUT_FILETYPE:-netcdf}
+
+export FIT_DIR=${ARCDIR}/fits
+[[ ! -d "${FIT_DIR}" ]] && mkdir -p "${FIT_DIR}"
+export HORZ_DIR=${ARCDIR}/horiz
+[[ ! -d "${HORZ_DIR}" ]] && mkdir -p "${HORZ_DIR}"
+export COMLOX=${DATA}/fitx
+[[ ! -d "${COMLOX}" ]] && mkdir -p "${COMLOX}"
+
+echo "echo err_chk">${DATA}/err_chk; chmod 755 ${DATA}/err_chk
+echo "echo postmsg">${DATA}/postmsg; chmod 755 ${DATA}/postmsg
+
+##############################################
+# Check spinup and available inputs
+##############################################
+
+if [[ ${xdate} -gt ${SDATE} ]]; then
+  for file in ${PRPI} ${sig1} ${sfc1} ${CNVS}; do
+    if [ ! -f ${file} ]; then
+      echo "FATAL ERROR: FILE MISSING: ${file}"
+      exit 1
+    fi
+  done
+
+  ##############################################
+  # RUN FIT2OBS VERIFICATION
+  ##############################################
+
+  "${fitdir}/batrun/excfs_gdas_vrfyfits.sh.ecf"
+  status=$?
+  [[ ${status} -ne 0 ]] && exit "${status}"
+
+  ##############################################
+  # End JOB SPECIFIC work
+  ##############################################
+
+  ##############################################
+  # Final processing
+  ##############################################
+  if [[ -e "${pgmout}" ]] ; then
+    cat "${pgmout}"
+  fi
+
+else
+
+  echo "Too early for FIT2OBS to run. Exiting."
+
```

Comment:
I would flip this other way around and right up front after `xdate` is calculated like so:

```bash
if [[ "${xdate}" -le "${SDATE:-}" ]]; then
    echo "Too early for FIT2OBS to run. Exiting."
fi
``` 

And then here, you can remove the big `if-block`

Reply:
This would need an `exit 0` too, right? If we exit right here then the `DATAROOT` that is created by `jjob_header.sh` at the start of the script won't be removed at the end of `JGDAS_FIT2OBS`. That's why I set up the if-block like I did.